src/Form/matching.js

```diff
@@ -968,8 +970,15 @@ const getLargestMeaningfulContainer = (el, form, cssSelector) => {
     const inputsInParentsScope = parentElement.querySelectorAll(cssSelector)
     // To avoid noise, ensure that our input is the only in scope
     if (inputsInParentsScope.length === 1) {
+        // If the parent has only 1 input and a label with text, we've found our meaningful container
+        const labelInParentScope = parentElement.querySelector('label')
+        if (labelInParentScope?.textContent?.trim()) {
+            return parentElement
+        }
```

Comment:
Hopefully, the comment says it all. Basically, before this, the algo would keep traversing up the DOM even if the markup looked like this:

```html
<div>
  <label>Username (note there is no for attribute)</label>
  <input id=username />
</div>
```